tdp/cli/commands/vars/edit.py

```diff
@@ -0,0 +1,295 @@
+            logger.error(str(strErr))
+            raise click.ClickException(strErr)
+
+    if not vars.exists():
+        raise click.BadParameter(f"{vars} does not exist.")
+    dag = Dag(collections)
+
+    # collections: instance of Collections.
+    # collections: "Collections",
+
+    # tdp_vars: Path to the tdp_vars directory.
+    # tdp_vars: Union[str, os.PathLike],
+
+    # repository_class: instance of the type of Repository used.
+    # repository_class: "Repository" = GitRepository
+
+    # service : exemple hdfs
+    # service : str
+
+    # recuperation du chemin du service dans le repo git
+    # service_tdp_vars = tdp_vars / service
+    # service_tdp_vars = vars / service_name
+
+    # recuperation du service variables
+    # repo = repository_class.init(service_tdp_vars)
+    repo = GitRepository(service_tdp_vars)
+    schemas = collections.get_service_schema(service_name)
+    service_variables = ServiceVariables(service_name, repo, schemas)
+
+    program_exit = False
+    file_name_is_validate = True
+
+    logger.debug(f"Writting in the service_component_file {service_component_file}")
+    # Loop until plan has no error or user
+    while not program_exit:
+        logger.debug(program_exit)
+        click.edit(
+            editor=os.environ.get("VISUAL", "") or os.environ.get("EDITOR", "") or "vi",
+            filename=str(service_component_file),
+        )
+
+        # Workaround to wait for the editor like vscode to be closed
+        click.pause("\nPress a key when you have finished editing the variables file")
+
+        try:
+            # update variables file
+            file_name_is_validate = False
+            strMsg = f"\n\nSchema structure validation for file_name: {service_component_file_shortname}\n"
+            logger.debug(str(strMsg))
+            service_variables.validate()
+            strMsg = f"\nValid schema structure for file_name: {service_component_file_shortname}\n"
+            click.echo(strMsg)
+
+            file_name_is_validate = True
+
+            # git commit of variables file modifications
+            try:
+                with repo.validate(commit_message) as repository:
+                    strMsg = f"Commit file: {service_component_file_shortname} modifications for the service_name {service_name}"
+                    logger.info(strMsg)
+                    repository.add_for_validation(service_component_file_shortname)
+
+                strMsg = f"\nCommit variables file: {service_component_file_shortname} modifications successful.\n"
+                click.echo(click.style(strMsg, fg="green", bold=True))
+
+            except EmptyCommit as e:
+                strMsg = f"Unmodify file {service_component_file_shortname} : {str(e)}"
+                logger.warning(strMsg)
+                strMsg = f"\nUnmodify file {service_component_file_shortname}\n"
+                click.secho(strMsg, fg="yellow", dim=True)
```

Comment:
You should (and can) avoid nesting try/except.

Reply:
This is now only one try/except.
The repository commit in now in a function.